test/mem2reg.cl

```diff
@@ -0,0 +1,11 @@
+// RUN: %clang_cc1 -O0 -S -triple spir-unknown-unknown -cl-std=CL2.0 -x cl -disable-O0-optnone %s -emit-llvm-bc -o %t.bc
+// RUN: llvm-spirv -s %t.bc
+// RUN: llvm-dis < %t.bc | FileCheck %s --check-prefixes=CHECK,CHECK-W
+// RUN: llvm-spirv -s -spirv-mem2reg %t.bc -o %t.opt.bc
+// RUN: llvm-dis < %t.opt.bc | FileCheck %s --check-prefixes=CHECK,CHECK-WO
```

Comment:
It is probably better swap names `CHECK-W` and `CHECK-WO`, i.e. `WO` stands for "without mem2reg", `W` stands for "with mem2reg"